x-pack/plugins/observability/public/embeddable/slo/alerts/components/slo_alerts_table.tsx

```diff
@@ -66,25 +75,34 @@ export const useSloAlertsQuery = (slos: SloItem[], timeRange: TimeRange) => {
           },
           {
             bool: {
-              should: slos.map((slo) => getSloInstanceFilter(slo.id, slo.instanceId)),
+              should: slos.map((slo) =>
+                getSloInstanceFilter(slo.id, slo.instanceId, showAllGroupByInstances)
+              ),
             },
           },
         ],
       },
     };
 
     return query;
-  }, [slos, timeRange]);
+  }, [showAllGroupByInstances, slos, timeRange.from]);
```

Comment:
Not related to current changes, but I was wondering why we don't have the timeRange.to in the range query. If user selects to see alerts from a specific date to another date, then we don't take the `to` into consideration. Do we need to create a ticket for this?

```
range: {
              '@timestamp': {
                gte: timeRange.from,
              },
            },
```

Reply:
@shahzad31 If user selects `to`  time range in the timepicker, it won't be applied right? Shouldn't the range contain the timeRange.to as well? Do you think we can do this in this PR? Otherwise a follow up PR is fine as well